cmd/restic/exclude.go

```diff
@@ -213,57 +218,117 @@ func gatherDevices(items []string) (deviceMap map[string]uint64, err error) {
 		if err != nil {
 			return nil, err
 		}
+
 		id, err := fs.DeviceID(fi)
 		if err != nil {
 			return nil, err
 		}
+
 		deviceMap[item] = id
 	}
+
 	if len(deviceMap) == 0 {
 		return nil, errors.New("zero allowed devices")
 	}
+
 	return deviceMap, nil
 }
 
+// IsAllowed returns true if the path is located on an allowed device.
+func (m DeviceMap) IsAllowed(item string, deviceID uint64) (bool, error) {
+	for dir := item; ; dir = filepath.Dir(dir) {
```

Comment:
Here we loop over parent dirs until one of those exactly matches one of the paths given to backup (and check if the device id matches).
Maybe it is clearer to loop over the paths to backup and check if `item` is a subdir (and then check for matching device ids). To test whether it's a subdir I used `filepath.Match` in #3121.

But I don't know which is more performant and as this will be called quite often, we should choose the more performant solution.

Reply:
I'd rather not change this code in this PR and postpone it for a later PR because I haven't changed it, that's the code from master:

https://github.com/restic/restic/blob/2f9346a5afa95a2a65c08a44117ace8808b62c1f/cmd/restic/exclude.go#L247-L264

Reply:
Yes, I saw that you just moved the code. Took me quite some time to understand what's going on, this is why I thought whether there is a clearer way. But the code is ok, of course! :+1: